### ISSUE TITLE
Multiple hosts - connection error

### ISSUE THREAD

**Issue by waszpaw** [anetrusov/ZabbixDBA#84]
Hello Alex,

Actually, I'm not sure if it's a bug. Situation:
* I have an array of hosts (proxies)
* One of them is not available (severe network infrastructure problem)
* Hosts not dependent on that particular proxy return connection error in the log files
* All the threads seem to be killed afterwards.

When I remove the host line responsible for the unavailable host and restart the zdba daemon, all comes back to normal.

Any suggestions?
BR
Pawel

**Comment by netrusov** [comment]
hi @waszpaw 

did you receive this message? https://github.com/anetrusov/ZabbixDBA/blob/master/lib/ZDBA/Sender.pm#L67

**Comment by waszpaw** [comment]
Seems so:

[2018/07/10 12:10:08][FATAL] [ZDBA::Sender::connect:67] unable to connect to 'A.B.C.D:10051': No route to host

PWA

**Comment by netrusov** [comment]
yes, that's the issue. what's expected behavior? just skip unavailable proxy?

**Comment by waszpaw** [comment]
For the moment configuration of proxies is in no-way related to the particular database instances.
In my case, I have one group of databases connected via one proxy and another one through another proxy (disjunctive sets).
I’d expect that I’d see communication errors only in these logs which belong to the related Oracle instances (as they are really unavailable).
Now, I think all my connections got kicked out, although half of them was perfectly doable.
PWA
